app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -673,5 +674,25 @@ def set_vm_interfaces_attributes(vm, vm_attrs)
       end
       vm_attrs
     end
+
+    def set_vm_volumes_attributes(vm, vm_attrs)
+      template = template(vm_attrs[:template]) if vm_attrs[:template]
+      template_volumes_names = template.volumes.map {|volume| volume.name } if defined? template.volumes
+      vm_attrs[:volumes_attributes]&.each do |key, volume|
+        unless template_volumes_names.include? volume[:name]
+          new_volume = {
```

Comment:
This is similar to the way the volume is built in `preallocate_and_clone_disks`, maybe some of it can be pulled out to a helper method?